src/main/java/com/byznass/tiolktrack/postgres/ConnectionFactory.java

```diff
@@ -0,0 +1,59 @@
+package com.byznass.tiolktrack.postgres;
+
+import com.byznass.tiolktrack.config.NoSuchPropertyException;
+import com.byznass.tiolktrack.config.PropertyProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static com.byznass.tiolktrack.config.Property.*;
+
+public class ConnectionFactory implements ConnectionProvider {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionFactory.class);
+	static final String URL = "jdbc:postgresql://%s/tiolktrack";
+
+	private final PropertyProvider propertyProvider;
+
+	@Inject
+	public ConnectionFactory(PropertyProvider propertyProvider) {
+
+		this.propertyProvider = propertyProvider;
+	}
+
+	@Override
+	public Connection getConnection() throws ConnectionFailureException {
+
+		try {
+			LOGGER.info("Trying to create a postgres connection");
+			Class.forName("org.postgresql.Driver");
+			Properties connectionDetails = getConnectionDetails();
+
+			String url = String.format(URL, propertyProvider.getValue(DB_HOST));
+			Connection connection =
+					DriverManager.getConnection(url, connectionDetails);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Connection" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3AS2095)